docs/Contributing.adoc

```diff
@@ -6,11 +6,38 @@ Herein lies the contribution guidelines for helping out with this project. Do ta
 
 === General rules
 
-* The Ansible <<coc,Code of Conduct>> still applies.
-* For git messages, branch names, ..., follow <<gitstyle,Git Style Guide>>
-* Should you wish to work on a completely new standard, GREAT, but please file an issue for tracking.
+* The Ansible link:https://docs.ansible.com/ansible/latest/community/code_of_conduct.html[Code of Conduct] still applies.
+* For git messages, branch names, ..., follow link:https://github.com/agis/git-style-guide[Git Style Guide].
+* Should you wish to work on a completely new standard, GREAT, but please file an **issue** for tracking. In your Pull Request, specify `closes #NUM` to automatically close the issue when the PR is merged.
 * To contribute, fork and make a pull request against the `development` branch.
-* All tasks should be in YAML literal.
+*  Use link:https://asciidoctor.org/docs/asciidoc-writers-guide/[asciidoc] for documentation.
+* Pull requests should only change one role, one config, or one role as applied to one config. There may be exceptions to this rule, but you must explain why a special exception should apply to a PR. Ask yourself, « Can this pull request be separeted in several smaller pull requests ? »
+* Pull Request titles:
+** Must start with `Add | Change | Fix | HOTFIX | Remove`
+** followed by `config | workload | role | core | test | documentation`
+** followed by the **name** of the config, workload, cloud provider, test or documentation
+** followed by a short description
+* Pull Request must be tested. Explain how you tested your change.
+* Owner of role or config must be identified.
+* If your Pull Request is not ready for review, open it as link:https://github.blog/2019-02-14-introducing-draft-pull-requests/[Draft]] or add `WIP` in the title.
+
+=== Code Quality Rules
+
+. a YAML `.yamllint` file should be added to every role and config when any substantial change is applied to the role or config, all new roles and configs must include a `.yamllint`.
+. All tasks should be in YAML literal.
+. Indentation is 2 white-spaces.
+. No in-line JSON format in Ansible
+. No use of `command` or `shell` module when specific Ansible module exists to do the same, without justification
+** If you have to, prefer `command` module to `shell` module when possible.
+** use `k8s` as much as possible and not `oc` command
+. All tasks must have names
+. All plays must have names
+. Roles must have README that document role variables.
+** In README, list and describe common variables and secrets
+. Roles must have `meta/main.yml` with author/contact information for support
+. Be extra careful with external dependencies. Spot them (external github repos, libs, etc) and make sure the versions are **pinned** (use versions, git tags, or commit ID).
+. In a role, ensure all variables names are prefixed with the role name to avoid name collisions with other roles.
+. Do not add `ignore_errors` to a task without justification.
```

Comment:
Add: "Prefer use of `failed_when` if a condition is not an error."

Reply:
done